pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -356,9 +358,11 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 			log.Debugf("Endpoint %v not found, skipping stale endpoint", key)
 			return
 		}
-		c.queue.Push(func() error {
-			return c.endpoints.onEvent(item, model.EventUpdate)
-		})
+		if shouldEnqueue("Pods", c.beginSync) {
```

Comment:
why not do this at start of the function?

Reply:
Also this function is not called on any pod, just ones that are in needResync, so I am not sure its actually something we want to skip

Reply:
This is what i thought: in k8s, for normal endpoint and pods, the pod creation must be before the endpoints, endpoints-controller list-watch pods and then reconciles the endpoints. It must be that during startup, istiod list-watch pods/endpoints, after cache syncs, for all the endpoints we can find the related pods. 

The issue why endpoint event occur before pods, is that there are two streams on flying.